### ISSUE TITLE
sweep x0 and xf during training

### ISSUE THREAD

**Issue by jsalfity** [jsalfity/sensitivity_games#7]
I'd like for `x0` and `xf` to change during the training process. Intuition says not much will change in long time horizons, because looking at the DARE calculation there is no usage of `x0` and `xf`. So I expect `K` to converge to the same result, regardless of `x0` and `xf`. In fact, `K` might converge faster if there are more variations. 

I see a problem with the way we are calculating the trajectory costs (and solving this may allow a time varying `xf` as well)/  The way we have defined the TrajectoryCost function, it's dependent on `xf`, which of course is good, but during training and updating it should be updating `xf`. So we need a method to update and/or redefine the `state_costs`.
https://github.com/jsalfity/sensitivity_games/blob/main/sensitivity_games/experiments/point_mass_regulation_train.py#L49-L53 and https://github.com/jsalfity/sensitivity_games/blob/main/sensitivity_games/trajectory_cost.py

Then another problem is updating `xf` in the controller. Again, we'd need a function to update `xf`.
https://github.com/jsalfity/sensitivity_games/blob/main/sensitivity_games/regulator_controller.py 

Seems like that would do it, so long as the `run_experiment()` function incorporates the additions/changes correctly, https://github.com/jsalfity/sensitivity_games/blob/main/sensitivity_games/experiments/point_mass_regulation_train.py#L46-L64. Important to note that `dynamics` and `controller` can't be recreated every epoch, because the torch variables are in `dynamics.theta` and `controller.K`.

It seems like things are getting messy. Wondering if there is a better way or time to refactor ?
Opened up a new branch, `sweep_initial_final_states_during_training`

**Comment by dfridovi** [comment]
I actually think there's no point in changing `xf` during training as long as the structure of the controller is `-K * (x - xf)`, i.e., as long as the controller already knows `xf` and it's not being learned. That is, what's happening in that control law is that we're already computing the correct tracking error and the learning problem is thus "normalized" or at least "centered." It could totally make sense to vary the initial state randomly, but that should have nothing to do with the structure of the controller. Am I missing something here?

**Comment by dfridovi** [comment]
Oh also, pro tip, standard git branch naming conventions are to prefix names with something meaningful, e.g. in a shared repo something like `<git id>/<short name>` or if you're basically the only contributor something like `feature/<name>` or `fix/<bug>` or `example/<blah>`

**Comment by jsalfity** [comment]
Ah right, that makes sense. Due to the subtraction, I think we only need to change one of them. But it does make sense to me that one has to change to provide the loss function with more data points. Let me give that a shot and see what happens.
